### PR TITLE
move row up/down for null metric ordering

### PR DESCRIPTION
**Pull request by asifmansoora**
when metrics sorted with null value - druid breaks
with this PR - we can check for nulls on left and right and move the row up/down accordingly

### CODE REVIEW

File: processing/src/main/java/io/druid/query/groupby/orderby/DefaultLimitSpec.java [hunk 1]
```diff
@@ -255,6 +255,13 @@ public int compare(Row left, Row right)
       @Override
```
Comment (leventov) [inline processing/src/main/java/io/druid/query/groupby/orderby/DefaultLimitSpec.java]: That whole ordering is `Comparator.comparing(row -> row.getRaw(column), Comparators.naturalNullsFirst())`

**Comment by himanshug** [comment]
:+1: after https://github.com/druid-io/druid/pull/4681#pullrequestreview-55824893 is resolved.

**Comment by himanshug** [comment]
@asifmansoora works at Yahoo and will be added to our cla.

File: processing/src/main/java/io/druid/query/groupby/orderby/DefaultLimitSpec.java (hunk 1 above)
Comment (himanshug) [inline processing/src/main/java/io/druid/query/groupby/orderby/DefaultLimitSpec.java]: probly meant `Ordering.from(Comparator.comparing(row -> row.getRaw(column), Ordering.from(comparator).nullsFirst()));`

Comment (leventov) [inline processing/src/main/java/io/druid/query/groupby/orderby/DefaultLimitSpec.java]: Then `Comparator.nullsFirst(comparator)`, this method is added in Java 8

Comment (asifmansoora) [inline processing/src/main/java/io/druid/query/groupby/orderby/DefaultLimitSpec.java]: but still `row` is not visible here `Comparator.comparing(row -> row.getRaw(column)`

Comment (leventov) [inline processing/src/main/java/io/druid/query/groupby/orderby/DefaultLimitSpec.java]: @asifmansoora unclear what do you mean

Comment (asifmansoora) [inline processing/src/main/java/io/druid/query/groupby/orderby/DefaultLimitSpec.java]: @leventov never mind, I wasnt able to get the `Row` reference inside comparing block in the first pass. But type casted it

File: processing/src/main/java/io/druid/query/groupby/orderby/DefaultLimitSpec.java [hunk 2]
```diff
@@ -249,15 +249,7 @@ public int compare(Row left, Row right)
...
-      @Override
-      public int compare(Row left, Row right)
-      {
-        return comparator.compare(left.getRaw(column), right.getRaw(column));
-      }
-    };
+    return Ordering.from(Comparator.comparing((Row row) -> row.getRaw(column), Comparator.nullsFirst(comparator)));
   }
 
   private Ordering<Row> dimensionOrdering(final String dimension, final StringComparator comparator)
```
Comment (leventov) [inline processing/src/main/java/io/druid/query/groupby/orderby/DefaultLimitSpec.java]: Could you please change `dimensionOrdering()` to align with `metricOrdering()`

**Comment by asifmansoora** [comment]
addressed changes.. also had nulls at last instead of being at first.

**Comment by asifmansoora** [comment]
@leventov could please review again ? Thanks a lot

**Pull request merged**